src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -24,27 +25,57 @@ class GitVersionPlugin implements Plugin<Project> {
 
     // Gradle returns 'unspecified' when no version is set
     private static final String UNSPECIFIED_VERSION = 'unspecified'
+    private DescribeCommand describeCommand = null
+    private Git git;
 
     void apply(Project project) {
         project.ext.gitVersion = {
+            doDescribe(project)
+        }
+
+        project.ext.gitVersionDescriber = {
```

Comment:
This should really be part of a configuration block, and the lambda should follow a prototype of `(Git) -> String`.